qsirecon/utils/bids.py

```diff
@@ -141,59 +142,92 @@ def collect_participants(bids_dir, participant_label=None, strict=False, bids_va
 
 
-def write_derivative_description(bids_dir, deriv_dir):
+def write_derivative_description(bids_dir, deriv_dir, dataset_links=None):
     from qsirecon import __version__
 
-    DOWNLOAD_URL = f"https://github.com/PennLINC/qsirecon/archive/{__version__}.tar.gz"
+    # Keys deriving from source dataset
+    orig_dset_description = os.path.join(bids_dir, "dataset_description.json")
+    if os.path.exists(orig_dset_description):
+        with open(orig_dset_description) as fobj:
+            dset_desc = json.load(fobj)
+    else:
+        config.loggers.utils.warning(f"Dataset description DNE: {orig_dset_description}")
+        dset_desc = {}
+
+    # Check if the dataset type is derivative
+    if "DatasetType" not in dset_desc.keys():
+        config.loggers.utils.warning(
+            f"DatasetType key not in {orig_dset_description}. Assuming 'derivative'."
+        )
+        dset_desc["DatasetType"] = "derivative"
+
+    if dset_desc.get("DatasetType", "derivative") != "derivative":
+        raise ValueError(
+            f"DatasetType key in {orig_dset_description} is not 'derivative'. "
+            "QSIRecon only works on derivative datasets."
+        )
 
-    desc = {
-        "Name": "qsirecon output",
-        "BIDSVersion": "1.1.1",
-        "PipelineDescription": {
+    dset_desc["Name"] = "QSIRecon output"
+    DOWNLOAD_URL = f"https://github.com/PennLINC/qsirecon/archive/{__version__}.tar.gz"
+    generated_by = dset_desc.get("GeneratedBy", [])
+    generated_by.insert(
+        0,
+        {
             "Name": "qsirecon",
             "Version": __version__,
             "CodeURL": DOWNLOAD_URL,
         },
-        "GeneratedBy": [
-            {
-                "Name": "qsirecon",
-                "Version": __version__,
-                "CodeURL": DOWNLOAD_URL,
-            }
-        ],
-        "CodeURL": "https://github.com/pennlinc/qsirecon",
-        "HowToAcknowledge": "Please cite our paper "
-        "(https://www.nature.com/articles/s41592-021-01185-5#citeas), and "
-        "include the generated citation boilerplate within the Methods "
-        "section of the text.",
-    }
+    )
+    dset_desc["GeneratedBy"] = generated_by
+    dset_desc["HowToAcknowledge"] = "Include the generated boilerplate in the methods section."
 
     # Keys that can only be set by environment
-    if "QSIRECON_DOCKER_TAG" in os.environ:
-        desc["DockerHubContainerTag"] = os.environ["QSIRECON_DOCKER_TAG"]
-    if "QSIRECON_SINGULARITY_URL" in os.environ:
-        singularity_url = os.environ["QSIRECON_SINGULARITY_URL"]
-        desc["SingularityContainerURL"] = singularity_url
-        try:
-            desc["SingularityContainerMD5"] = _get_shub_version(singularity_url)
-        except ValueError:
-            pass
+    if "FMRIPREP_DOCKER_TAG" in os.environ:
+        dset_desc["GeneratedBy"][0]["Container"] = {
+            "Type": "docker",
+            "Tag": f"nipreps/fmriprep:{os.environ['FMRIPREP_DOCKER_TAG']}",
```

Comment:
```suggestion
            "Tag": f"pennlinc/qsirecon:{os.environ['QSIPREP_DOCKER_TAG']}",
```

Reply:
I'm actually not sure whether these should be qsiprep or qsirecon - is it for the input data or the data processed here?

Reply:
It's for the environment used by QSIRecon. I'll fix it.